src/core.ts

```diff
@@ -124,6 +132,30 @@ export function instrument(eventState: IInstrumentationEventState) {
   }
 }
 
+export function onWorkerMessage(evt: MessageEvent) {
+  if (state !== State.Unloaded) {
+    let message = evt.data;
+    switch (message.type) {
+      case WorkerMessageType.Upload:
+        let uploadMsg = message as IUploadMessage;
+        let onSuccess = (status: number) => { mapProperties(droppedPayloads, uploadDroppedPayloadsMappingFunction, true); };
+        let onFailure = (status: number) => { onFirstSendDeliveryFailure(status, uploadMsg.rawData, uploadMsg.compressedData); };
+        upload(uploadMsg.compressedData, onSuccess, onFailure);
+        pendingEvents.splice(0, uploadMsg.eventCount);
```

Comment:
>     pendingEvents.splice(0, uploadMsg.eventCount); [](start = 3, length = 51)

Are you guaranteed that the state of this will always match the messages back from the worker? Seems a bit brittle. I would at least add some checking here that they align.

Reply:
Yes, as long as there is only one worker. The order of messages is guaranteed, so worker will be compressing and sending back events in the exact same order as pushed them on the queue.

---
In reply to: [137240383](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137240383) [](ancestors = 137240383)

Reply:
Can you add that as comment? 

---
In reply to: [137355388](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137355388) [](ancestors = 137355388,137240383)

Reply:
Added

---
In reply to: [137421997](https://github.com/Microsoft/clarity-js/pull/28#discussion_r137421997) [](ancestors = 137421997,137355388,137240383)